src/components/Login/Login.js

```diff
@@ -1,77 +1,91 @@
 import styled from 'styled-components/macro';
-import { ButtonWithIcon } from 'components';
+import { useEffect, useState } from 'react';
+import { TextInput, ButtonWithIcon, PasswordInput } from 'components';
 
-export const Login = onSubmit => {
-    function login(event) {
-        event.preventDefault();
-        const form = event.target;
-        const name = form.name.value;
-        if (name.indexOf('.') !== -1 && name.indexOf('@') !== -1) {
-            onSubmit(name);
-            form.reset();
+export const Login = ({ loginHandler }) => {
+    const dataNotFound = false;
+    const [nameFilled, setNameFilled] = useState(false);
+    const [nameWrongFormat, setNameWrongFormat] = useState(false);
+    const [passwordWrongLength, setPasswordWrongLength] = useState(false);
+    const [passwordFilled, setPasswordFilled] = useState(false);
+    const [inactive, setInactive] = useState(true);
+
+    const focusOutName = event => {
+        const input = event.target.value;
+        if (
+            input &&
+            input.indexOf('.') !== -1 &&
+            input.indexOf('@') !== -1 &&
+            input.indexOf('@') > input.indexOf('.') + 1 &&
+            input.indexOf('@') + 1 < input.length
+        ) {
+            setNameFilled(true);
+            setNameWrongFormat(false);
+        } else {
+            setNameFilled(false);
+            setNameWrongFormat(true);
+        }
+    };
+
+    const focusOutPassword = event => {
+        const input = event.target.value;
+        if (input.length >= 6) {
+            setPasswordFilled(true);
+            setPasswordWrongLength(false);
+        } else {
+            setPasswordFilled(false);
+            setPasswordWrongLength(true);
+        }
+    };
+
+    useEffect(() => {
+        if (nameFilled === true && passwordFilled === true) {
+            setInactive(false);
+        } else {
+            setInactive(true);
         }
-    }
+    }, [passwordFilled, nameFilled]);
```

Comment:
This will trigger a re-render after mount because it will update a state, even if it's not required.

```jsx
useEffect(() => {
        if (nameFilled === true && passwordFilled === true && inactive) {
            setInactive(false);
        } else if (!inactive) {
            setInactive(true);
        }
    }
    }, [inactive, passwordFilled, nameFilled]);
```